Assets/MixedRealityToolkit.SDK/Experimental/Features/Dwell/DwellHandler.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Dwell
+{
+    [Serializable]
+    public class PointerUnityEvent : UnityEvent<IMixedRealityPointer>
+    {
+    }
+
+    /// <summary>
+    /// Use this component to add a Dwell modality (https://docs.microsoft.com/en-us/windows/mixed-reality/gaze-and-dwell) to the UI target.
+    /// </summary>
+    public class DwellHandler : MonoBehaviour, IMixedRealityFocusHandler
+    {
+        protected enum DwellStateType
+        {
+            None = 0,
+            FocusGained,
+            DwellIntended,
+            DwellStarted,
+            DwellCompleted,
+            DwellCanceled,
+        }
+
+        [Header("Dwell Settings")]
+        [SerializeField]
+        protected DwellProfile dwellProfile = null;
+
+        [Header("Dwell Events")]
+        [SerializeField]
+        private PointerUnityEvent DwellIntended = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellStarted = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellCompleted = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellCanceled = new PointerUnityEvent();
+
+        /// <summary>
+        /// Invoke this event if the dwell is interrupted with alternate modality
+        /// </summary>
+        public UnityAction CancelDwellAction;
+
+        /// <summary>
+        /// Property exposing the computation for what percentage of dwell has progressed.
+        /// </summary>
+        public virtual float DwellProgress
+        {
+            get
+            {
+                switch (CurrentDwellState)
+                {
+                    case DwellStateType.None:
+                    case DwellStateType.FocusGained:
+                        return 0;
+                    case DwellStateType.DwellStarted:
+                        return Mathf.Clamp((float)(dwellProfile.TimeToCompleteDwell - FillTimer) / dwellProfile.TimeToCompleteDwell, 0f, 1f);
```

Comment:
>Mathf.Clamp((float)(dwellProfile.TimeToCompleteDwell - FillTimer) / dwellProfile.TimeToCompleteDwell, 0f, 1f); [](start = 31, length = 110)

Nit: Make this a private helper (because it's used twice in the context of this getter)